notebook-lessons/oss-module-01-git.ipynb

```diff
@@ -2709,6 +2712,246 @@
+    "\n",
+    "### Why This Approach Works Best\n",
+    "\n",
+    "**DCO signoff is harmless on projects that don't require it** - it's just an extra line in your commit message that proves you wrote the code. Think of it as additional documentation that you have the rights to contribute the code. Projects that don't require DCO will simply ignore the `Signed-off-by` line.\n",
+    "\n",
+    "**Benefits of using both by default:**\n",
+    "- 🎯 **One setup, works everywhere** - No need to remember different commands for different projects\n",
+    "- 🛡️ **Maximum compliance** - Covers both legal (DCO) and technical (signing) requirements\n",
+    "- 📝 **Better documentation** - Shows clear authorship and legal attestation\n",
+    "- ⚡ **Simple workflow** - Just use `git commit -m \"message\"` everywhere\n",
+    "\n",
+    "### Alternative: Project-Specific Aliases\n",
+    "\n",
+    "If you prefer more granular control, you can set up specific aliases:\n",
+    "\n",
+    "```bash\n",
+    "# For DCO signoff only\n",
+    "git config --global alias.dco 'commit --signoff'\n",
+    "\n",
+    "# For signing only  \n",
+    "git config --global alias.sign 'commit --gpg-sign'\n",
+    "\n",
+    "# For both (recommended)\n",
+    "git config --global alias.secure 'commit --gpg-sign --signoff'\n",
+    "```\n",
+    "\n",
+    "Then use:\n",
+    "```bash\n",
+    "git dco -m \"Your message\"     # DCO only\n",
+    "git sign -m \"Your message\"    # Signing only\n",
+    "git secure -m \"Your message\"  # Both\n",
+    "```\n",
+    "\n",
+    "### Recommended Setup Commands\n",
+    "\n",
+    "**For the \"one command for everything\" approach (recommended):**\n",
+    "\n",
+    "```bash\n",
+    "# Set up automatic signing\n",
+    "git config --global gpg.format ssh\n",
+    "git config --global user.signingkey ~/.ssh/id_ed25519.pub\n",
+    "git config --global commit.gpgsign true\n",
+    "\n",
+    "# Override commit to always include DCO signoff\n",
+    "git config --global alias.commit 'commit --signoff'\n",
+    "```\n",
+    "\n",
+    "Now `git commit -m \"Your message\"` will automatically include both DCO signoff and cryptographic signing!\n",
+    "\n",
+    "### Important Notes\n",
+    "\n",
+    "1. **Automatic Signing**: Setting `commit.gpgsign true` means **all commits are automatically signed**. You don't need to remember any flags.\n",
+    "\n",
+    "2. **DCO Signoff is Always Safe**: Adding DCO signoff to all commits doesn't hurt - it's just extra documentation that you have rights to contribute the code.\n",
+    "\n",
+    "3. **Repository-Specific Settings**: If you don't want to make these changes globally, omit the `--global` flag and run these commands in each repo where they should apply.\n",
+    "\n",
+    "4. **Fixing Commits**: If you need to fix a commit:\n",
+    "   ```bash\n",
+    "   git commit --amend -m \"Updated message\"  # Uses your existing config\n",
```

Comment:
Either omit this or mention that `git commit --amend` is recommended only for unpublished changes (i.e. before you've pushed the changes)